### ISSUE TITLE
hits process limit

### ISSUE THREAD

**Issue by stbuehler** [ahupowerdns/metronome#22]
metronome forks a connection for each carbon connection, which often seem to never close (had over 600 threads running for 3 pdns sources)

also it reads only single bytes from network due to unbuffered line reading (check with strace).

**Comment by ahupowerdns** [comment]
This is two tickets. The first one, we don't have that here, but we know
there are some issues with forks and threads in std::thread on some
platforms. Compiler & OS please?

**Comment by stbuehler** [comment]
debian jessie, gcc-4.9; I said "fork", but I guess you could argue I should have set "clone" - but the problem is that it counts to the nproc limit anyway. Forking or cloning per connection is just not acceptable in professional environments imho - easy DoS.
I don't think this is an issues related to forks and std::thread (when I attached strace all those threads had a read() syscall pending; not sure why the connections didn't get closed).

I can't test further because I moved to a real graphite setup (needed for pdnscontrol anyway).

**Comment by ahupowerdns** [comment]
No, we also clean up threads, so that does not fit your original problem
description. If you see 600 threads and not 600 connections, something is
wrong.

I see 1 thread per connection here. ps -eLf | grep metronome and netstat
-anp | grep metronome agree.

Regarding "not being professional", thank you for that. I have no issues
with cloning for a connection for metronome. 

	Bert
